From 21bed0adb9915759937d113ebbe0fc311cdd782b Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 28 May 2019 15:00:32 +0300
Subject: [PATCH] [SW-4185] pll: don't skip clk switch prior pll cfg

Do not skip the divisor output clock select moving to reference clock
before PLL configuration.

The HL_PLL_SKIP is used to skip the divisor output clock select
configuration, but it is not checked while configuring the PLL core.
This effectively allows the "skipped" divider output clock select to
output clock from PLL while it is being configured/reset/etc.
This scenario is not good and should be avoided.

Change-Id: I135c8311e8d029e9c279d6834f05f58ec2a05517
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index 8b427b7..7be7e70 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -77,9 +77,6 @@
 	u64 reg;
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
-		if (cfg->div_sel[sel] == HL_PLL_SKIP)
-			continue;
-
 		reg = cfg->base + HL_PLL_DIV_SEL + (sel * sizeof(u32));
 		switch (regmap_readl(reg)) {
 		case HL_PLL_DIV_PLL:
